internal/manager/filter/filter.go

```diff
@@ -0,0 +1,25 @@
+package filter
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/controller"
+)
+
+// CreateSingleResourceFilter creates a filter function that filters out all resources except the one
+// with the given namespaced name.
+func CreateSingleResourceFilter(targetNsName types.NamespacedName) controller.NamespacedNameFilterFunc {
+	return func(nsname types.NamespacedName) (shouldProcess bool, msg string) {
+		if nsname != targetNsName {
+			msg := fmt.Sprintf(
+				"Resource is ignored because this controller only supports a single resource %s/%s of that type",
```

Comment:
```suggestion
				"Resource %s/%s is ignored because this controller only supports a single resource of that type",
```

Reply:
this is intentional - we just mention the resource that the controller watches, not the one that is being ignored. It is assumed that the context about the resource being ignored will be provided by the reconciler -- https://github.com/nginxinc/nginx-kubernetes-gateway/blob/dc046cc58a97e04bace4646053d10e263f4b4f29/internal/controller/reconciler.go#L64-L65

Reply:
Oh I see